config/v2/resolvers.go

```diff
@@ -112,6 +114,36 @@ func ResolveAWSProvider(commons ...Common) *AWSProvider {
 	return nil
 }
 
+// ResolveBackend returns the Backend configuration for a given component, after applying all inheritance rules
+func ResolveBackend(commons ...Common) *Backend {
+	kind := lastNonNil(BackendKindGetter, commons...)
+
+	// This feels like a somewhat hacky way to do this, but not sure of a better place yet
+	if kind == nil {
```

Comment:
another random idea is to have a custom marshal on the backend struct that sets the s3 default and then calls the normal unmarshal (which could override the field)

Reply:
The problem with that is if you have inheritance from A->B and set something explicitly in A, but then our marshal code injects something in B, then you resolution will be wrong.

I don't think you can do this at the config / parsing level at all for that reason.

Reply:
ahh, good catch